website/blog/2024-10-21-creating-an-extension.md

```diff
@@ -0,0 +1,74 @@
+
+## Introduction to Extensions
+
+Extensions are abundant in Podman Desktop and can be found in the **Extensions -> Catalog** section.
+
+![extension catalog](img/podman-desktop-release-1.10/extension-catalog.png)
+
+Each extension expands on Podman Desktop, such as providing [Kubernetes development clusters with Minikube](https://github.com/containers/podman-desktop-extension-minikube) or even [analyzing your images](https://github.com/containers/podman-desktop-extension-layers-explorer).
+
+Below we will go over how to start creating your first extension!
+
+## Getting Started with Your Project
+
+The first step in creating your extension is setting up the project environment. To learn how to configure the project and add basic components, check out the **[Templates for creating an extension](/docs/extensions/templates)** guide, which walks you through initializing your project from an official template.
+
+## Adding UI Components
+
+One of the most common tasks when creating an extension is enhancing the user interface. Whether it’s adding buttons, panels, or icons, UI components help make your extension more interactive and accessible. Learn more about this in the **[Adding UI components](/docs/extensions/developing/adding-ui-components)** documentation, where you’ll find instructions on creating and integrating components into the application’s UI.
+
+### Working with Icons
+
+Icons are a great way to make your extension more visually unique. You can learn how to add and style custom icons by following the **[Adding icons](/docs/extensions/developing/adding-icons)** documentation.
+
+### Menus and Navigation
+
+Extensions often integrate with existing menus and navigation to offer users easy access to new commands and features. If you want to add items to the context menu, explore the **[Menu configuration](/docs/extensions/developing/menu)** documentation, which explains how to add commands to menus and control when they are displayed using When Clauses.
+
+## Adding and Configuring Commands
+
+Commands are the backbone of most extensions, allowing users to interact with the application and trigger specific actions. 
+
+If you need to define and register custom commands, the **[Commands](/docs/extensions/developing/commands)** guide will show you how to create commands that respond to user actions or input, and tie them into your extension’s workflow.
+
+You can also configure these commands to appear in different contexts. Check out the **[When clause Contexts](/docs/extensions/developing/when-clause-context)** documentation to learn more about restricting commands to specific scenarios.
+
+## Setting Up Onboarding Workflows
+
+Creating a smooth onboarding experience is essential to help users get started with your extension. This includes steps for CLI binary installations or other initial setup values.
```

Comment:
A screenshot here again could be beneficial for the user to understand what can be done

Reply:
Added screenshot from our compose onboarding